### ISSUE TITLE
Grant-CsOnlineVoiceRoutingPolicy fails when being run by a guest user with Teams Administrator role

### ISSUE THREAD

**Issue by horsi3** [MicrosoftDocs/office-docs-powershell#9173]
MSP managing MSTeams for a client in their tenant. Previously we were using admin accounts local to their tenant but are now using B2B guest accounts which are assigned the Teams Administrator role.

Now when we provision Teams users for the client one command is failing "Grant-CsOnlineVoiceRoutingPolicy".
It returns the error attached.

![image](https://user-images.githubusercontent.com/102339348/160048585-ced7e0d2-df1c-4710-8f67-7e99ab59e43f.png)

Only one of our users is not affected, and the only difference with their account is they have the global administrator role. Obviously we cannot grant that level of access to our entire support team as a workaround.

If the affected users do the same thing via the Teams Admin Center, it works successfully even though using the same guest account.

We've confirmed both module versions 3.1.0 and 4.0.0 have this issue when using guest accounts in the tenant.

Thanks

**Comment by JensTrier** [comment]
@horsi3 thanks for the feedback. Please open a Microsoft support case and they will work with you to get this root caused.

**Comment by yogkumgit** [comment]
@horsi3 Hope this comment is helpful for you. Considering the nature of the issue, as advised, please open a service ticket in your tenant and follow with them for the resolution. We proceed to close it as here we are limited to documentation fixes and improvements only.
We will be very happy if you can share the outcome or resolution with us if you see documentation update is required. We will certainly update this documentation with that valuable information. Thanks again, for taking out some time to open the issue. Appreciate and encourage you to do the same in future also.

**Comment by horsi3** [comment]
We found the problem here. The affected users were logged into desktops with their desktop AD account. This account does not have local administrator privileges. When they open powershell, they were elevating to run as admin, using their other AD account which is for administrator use. This is required when installing modules from the gallery but obviously not required to actually run the commands in the module.
They started running powershell without elevating privileges and this resolved the issue.
I assume that I was not affected because my machine is not domain joined and my desktop accounts is the local administrator unlike the other support staff.
Seems very strange that the desktop user and the elevated user must match to run this particular command.

I did raise a MS ticket but we found the above resolution before they gave us a meaningful response. I've suggested they could possibly tidy up this behavior but otherwise to close the ticket as we have a solution regardless.
